src/api/AsyncStorage.js

```diff
@@ -4,8 +4,8 @@
 
 function wrap(value, callback) {
   return Promise.resolve(value).then(
-    obj => callback(null, obj),
-    err => callback(err)
+    obj => callback && callback(null, obj),
```

Comment:
this line should be
`obj => {callback && callback(null, obj); return obj;}`
otherwise `getItem` receives `undefined` if no callback is provided, or receives the return value from the callback if provided.